src/datepicker/hijri/ngb-calendar-islamic-umalqura.ts

```diff
@@ -0,0 +1,268 @@
+  '010011101101', '000101101101', '100010110110', '101010100110', '110101010010',
+  // 1525-1529
+  '110110101001', '010111010100', '101011011010', '100101011011', '010010101011',
+  // 1530-1534
+  '011001010011', '011100101001', '011101100010', '101110101001', '010110110010',
+  // 1535-1539
+  '101010110101', '010101010101', '101100100101', '110110010010', '111011001001',
+  // 1540-1544
+  '011011010010', '101011101001', '010101101011', '010010101011', '101001010101',
+  // 1545-1549
+  '110100101001', '110101010100', '110110101010', '100110110101', '010010111010',
+  // 1550-1554
+  '101000111011', '010010011011', '101001001101', '101010101010', '101011010101',
+  // 1555-1559
+  '001011011010', '100101011101', '010001011110', '101000101110', '110010011010',
+  // 1560-1564
+  '110101010101', '011010110010', '011010111001', '010010111010', '101001011101',
+  // 1565-1569
+  '010100101101', '101010010101', '101101010010', '101110101000', '101110110100',
+  // 1570-1574
+  '010110111001', '001011011010', '100101011010', '101101001010', '110110100100',
+  // 1575-1579
+  '111011010001', '011011101000', '101101101010', '010101101101', '010100110101',
+  // 1580-1584
+  '011010010101', '110101001010', '110110101000', '110111010100', '011011011010',
+  // 1585-1589
+  '010101011011', '001010011101', '011000101011', '101100010101', '101101001010',
+  // 1590-1594
+  '101110010101', '010110101010', '101010101110', '100100101110', '110010001111',
+  // 1595-1599
+  '010100100111', '011010010101', '011010101010', '101011010110', '010101011101',
+  // 1600
+  '001010011101'
+];
+
+function getDaysDiff(date1: Date, date2: Date): number {
+  const diff = Math.abs(date1.getTime() - date2.getTime());
+  return Math.round(diff / ONE_DAY);
+}
+
+@Injectable()
+export class NgbCalendarIslamicUmalqura extends NgbCalendarHijri {
+  /**
+  * Returns the equivalent islamic(Umalqura) date value for a give input Gregorian date.
+  * `gdate` is s JS Date to be converted to Hijri.
+  */
+  fromGregorian(gDate: Date): NgbDate {
+    let hDay = 1, hMonth = 0, hYear = 1300;
+    let daysDiff = getDaysDiff(gDate, GREGORIAN_FIRST_DATE);
+    if (gDate.getTime() - GREGORIAN_FIRST_DATE.getTime() >= 0 && gDate.getTime() - GREGORIAN_LAST_DATE.getTime() <= 0) {
+      let year = 1300;
+      for (let i = 0; i < MONTH_LENGTH.length; i++, year++) {
+        for (let j = 0; j < 12; j++) {
+          let numOfDays = parseInt(MONTH_LENGTH[i][j], 10) + 29;
+          if (daysDiff <= numOfDays) {
+            hDay = daysDiff + 1;
+            if (hDay > numOfDays) {
+              hDay = 1;
+              j++;
+            }
+            if (j > 11) {
+              j = 0;
+              year++;
+            }
+            hMonth = j;
+            hYear = year;
+            return new NgbDate(hYear, hMonth + 1, hDay);
+          }
+          daysDiff = daysDiff - numOfDays;
+        }
+      }
+    } else {
+      return new NgbCalendarIslamicCivil().fromGregorian(gDate);
```

Comment:
creating new calendar object every time you need to do the date conversion looks suspicious to me

Reply:
what about replacing that line with this 
`return NgbCalendarIslamicCivil.prototype.fromGregorian.call(this, gDate);` ?

Reply:
Or maybe just create a static instance of the `NgbCalendarIslamicCivil` and reference it?